lib/OpenQA/Schema/ResultSet/Users.pm

```diff
@@ -22,4 +22,6 @@ sub create_user ($self, $id, %attrs) {
     return $user;
 }
 
+sub system ($self) { $self->search({username => 'system', provider => ''}, {rows => 1})->first }
```

Comment:
Since there is a unique constraint on username + provider, likely `single` will be enough.
```
$self->single({username => 'system', provider => ''})
```

Reply:
So does that mean that

```perl
$self->single({username => 'system', provider => ''})
```

is equivalent to

```perl
$self->search({username => 'system', provider => ''}, {rows => 1})->first
```

and the same in all the other cases as well?

Reply:
Only if the arguments are a unique constraint. It doesn't add `rows => 1` for example, so the result must return 0 or 1 rows.

I was surprised that you didn't use `find` here.

Reply:
quote from the docs:
> Note                                                                            
> As of 0.08100, this method enforces the assumption that the preceding query **returns only one row**. If more than one row is returned, **you will receive a warning**:
>
> `Query returned more than one row ` 
> In this case, you should be using "next" or "find" instead, or if you really know what you are doing, use the "rows" attribute to explicitly limit the size of the resultset.